pkg/plugins/manager/loader/finder/local.go

```diff
@@ -182,25 +183,42 @@ func (l *Local) getAbsPluginJSONPaths(path string, followDistFolder bool) ([]str
 				return fmt.Errorf("filepath.Walk reported an error for %q: %w", currentPath, err)
 			}
 
-			if fi.Name() == "node_modules" {
+			switch {
+			case fi.Name() == "node_modules":
 				return util.ErrWalkSkipDir
-			}
-
-			if fi.IsDir() {
+			case fi.Name() == "dist" && !followDistFolder:
+				return util.ErrWalkSkipDir
+			case fi.IsDir():
 				return nil
-			}
-
-			if fi.Name() != "plugin.json" {
+			case fi.Name() != "plugin.json":
 				return nil
 			}
 
-			pluginJSONPaths = append(pluginJSONPaths, currentPath)
+			pluginJSONPaths[currentPath] = struct{}{}
 			return nil
 		}); err != nil {
 		return []string{}, err
 	}
 
-	return pluginJSONPaths, nil
+	// For external plugins, if we found a plugin directory that contains a "dist" directory, we want to ignore the
+	// plugin.json file found in the parent folder if it exists.
+	// For example, if we found encountered /path/to/plugin/dist/plugin.json, we want to ignore the path
+	// /path/to/plugin/plugin.json in order to prioritize the built version and avoid loading the plugin twice.
```

Comment:
I'm not sure if we need to cover more cases here such as:

- /myPlugins/grafana-foo-datasource/src/plugin.json
- /myPlugins/grafana-foo-datasource/dist/plugin.json